src/app.rs

```diff
@@ -168,8 +168,8 @@ where
         }
     }
     pub fn go_up(&mut self) {
-        if let Some(tile) = self.board.pop_previous_index() {
-            self.board.set_selected_index(&tile);
+        if let Some(index) = self.board.pop_previous_index() {
```

Comment:
Good catch. I was trying something different first and forgot to change back. ;)